src/Obsidian/Cache.ts

```diff
@@ -91,6 +92,20 @@ export function getTasksFromFileContent2(
                     ),
                     fallbackDate: dateFromFileName.value,
                 });
+
+                if (task !== null) {
+                    const parentTask: Task | null = line2Task.get(listItem.parent) ?? null;
```

Comment:
Because [ListItemCache](https://docs.obsidian.md/Reference/TypeScript+API/ListItemCache) allows parent values to be negative, I was a little nervous about this code... thinking that perhaps it worked because the first list item in all the tests being used was always on line 0, so it worked by luck.

So I changed one of the samples to this:

```
# Test heading

- [ ] #task parent task
    - [ ] #task child task 1
        - [ ] #task grandchild 1
    - [ ] #task child task 2
        - [ ] #task grandchild 2
- [ ] #task sibling
```

And once the inline snapshot was updated, the test still passed... even though 2 tasks have parent `-2`... 

But they are root-level tasks, so it's all OK.

Maybe a comment would be helpful, explaining why negative `.parent` is fine?

Reply:
fixed that. Added both test for non-first task and added a comment for possible negative listItem.parent value

Reply:
That comment was super clear - and thanks for adding the test too.